### ISSUE TITLE
Sentinel::check() always returns false for a logged user

### ISSUE THREAD

**Issue by paprilia** [cartalyst/sentinel#251]
This is the login method in `UserController`

```php
public function login(Request $request){
        $this->setHeaderRequest();

        $credentials =$request->only('email','password');

        $user = DB::table('users')->where('email', $request->input('email'))->get();

        if(count($user) > 0){
            try {

                //$status=Sentinel::authenticateAndRemember($credentials);
                $status=Sentinel::authenticate($credentials);

                if($status){
                    $this->_setSession();

                    $status=true;

                    //getting roles of group
                    $users=DB::table('users_group_roles')
                        ->where('i_group',Sentinel::check()->i_group)
                        ->get();

                    $message = 'Authenticated';

                }else{
                    $users=null;
                    $message = "We failed to authenticate your credentials. Please check your email or password.";
                }
            } catch (ThrottlingException $ex) {
                $users=null;
                $status=false;
                $message = $ex;
            }
        }else{
            $status = false;
            $users = null;
            $message = 'Sorry, you are not registered! Please register <a href="signup" class="signup-button text-black-small">here</a>';
        }


        $user=array(
            'data'      =>  Session::all(),
            'roles'     =>  $users,
            'status'    =>  $status,
            'message'   => $message
        );

        $this::printJSON($user);
    }
```

When I tried to debug the `Sentinel::check()` in this function, it indeed returns some data.

After the login is successful, user is redirected to a page managed by `AccountController`. Here is the function in that controller that shows the after login view :

```php
public function account(Request $request){
        if(!Session::has('is_logged') && !Session::get('is_logged')){
            return Redirect::to('login');
        }

        return view('frontend.account_'.$request->path(), ['subaccounts' => $this->subaccounts, 'menus' => $this->menus]);
    }
```

however when I tried to debug the `Sentinel::check()` it always returns false in this controller and other controllers despite the user is logged in. 

Does anyone know what causes the issue and how to solve it ?

**Comment by brunogaspar** [comment]
What Laravel version are you using?
Do sessions work properly on your app? As it seems seems are not being stored correctly.

**Comment by hizbul25** [comment]
I am using Sentinel with Laravel 5.3 .
Inside the login method

Sentinel::authenticate($credentials)

works fine and each time i get user object from Sentinel::check() after successful login.

But problem arise after redirect, and Sentinel::check() showing false inside the route, middle-ware, views. Any idea about the issue?

**Comment by brunogaspar** [comment]
@hizbul25 
- Make sure that you've followed the steps to install Sentinel properly;
- Make sure that you've the correct permissions, per Laravel documentation;
- Make sure that your routes are using the `web` middleware;
- Also check [this](https://laravel-news.com/controller-construct-session-changes-in-laravel-5-3) which is a common user error on Laravel 5.3+;

Other than that, we'll need to see reproducible code that would create the issue you're having.

**Comment by toddroper** [comment]
I'm having the same issue and the it is definitely inside Sentinel.  I've traced it back to the check method in the IlluminatePersistenceRepository.  The persistence code is saved to the DB and exists in the session yet this fails:
if ($code = $this->session->get()) {
            return $code;
}

There is an issue with how this logic is pulling the code.

Session:
array:5 [▼
  "_previous" => array:1 [▶]
  "_flash" => array:2 [▶]
  "_token" => "gnr0gSFjbXpH9GDGwazbtsxfr9cGr2y7HGVWbm8V"
  "sentinel" => "RxKUDcIGonsYSOtuCJk8aN6SjNZubwSP"
]

DB:
19	4	RxKUDcIGonsYSOtuCJk8aN6SjNZubwSP	2017-07-17 19:24:55	2017-07-17 19:24:55

Sentinel::check():
false

**Comment by toddroper** [comment]
I'll also add that I followed the Sentinel install instructions from the main site (https://cartalyst.com/manual/sentinel/2.0#laravel-5) and Sessions are working in all other parts of my code.

**Comment by brunogaspar** [comment]
@toddroper Without seeing any code it's difficult to figure out what is your issue.

The only thing that comes to my mind is [this](https://laravel-news.com/controller-construct-session-changes-in-laravel-5-3).

**Comment by toddroper** [comment]
I solved the problem, you need to tell users that report this issue to check which Facade they're using and inform them that the native session key should be left as 'cartalyst_sentinel'.  I updated the key and that caused the initial failure, reverting it fixed the problem.  The second failure was due to the Native facade being used in one Model.  I will say that you are less than helpful and I will not being using this library in future projects because you don't seem to understand your own code.

**Comment by brunogaspar** [comment]
> I will say that you are less than helpful and I will not being using this library in future projects because you don't seem to understand your own code.

Thank you.

**Comment by mdisconzi** [comment]
My problem was solved putting all routes "web", laravel 5.6
